### ISSUE TITLE
The badcase annotations

### ISSUE THREAD

**Issue by Karthik-Suresh93** [zhpmatrix/VisDrone2018#1]
Hi,

Could you please tell which annotations are causing nan problems? I tried two or three detectors, none of them are converging.

**Comment by zhpmatrix** [comment]
I have sent a mail to host, as follows, hope to help you.
        When I put the dataset into the pipeline, there are some NaN mistakes. I have spent lots of time on finding the bugs, the ground truth of boxes are maybe the devils. After filtering the annotations, I find three special annotation files with suspicious coordinates as follow.
        1.   0000293_03401_d_0000939.txt
        2.   9999985_00000_d_0000020.txt
        3.   9999999_00590_d_0000267.txt
        The first file with a line as bbox_left=1008, bbox_top=374, has bbox_height=0. That's to say, a line. The second file with a line as bbox_left=611, bbox_top=158 and the last file with a line as bbox_left=545, bbox_top=414 have same problems as the first file.
        The bboxes as lines aren't common, it will make many NaN errors, maybe just for me?
        After that, I check the bboxes in images, bboxes of above lines are separately ignored region, car, ignored region. Certainly, all of them are really small enough to delete them directly. I just do that to cancel the NaN mistakes, the model is training happily.
         Are they problems in datasets? TKS!

**Comment by Karthik-Suresh93** [comment]
Yes, I found exactly these annotation files when I checked too.  Also, if we use:

`  #make zero-based by subtracting 1`
` xmax = xmin + w -1` 
   ` ymax = ymin + h -1 `

instead of:
    `xmax = xmin + w`
    `ymax = ymin + h`

I found the follwoing additional bad annotations:
`problem_images=['0000182_01653_d_0000040.jpg',
 '0000293_03401_d_0000939.jpg',
 '9999945_00000_d_0000070.jpg',
 '9999955_00000_d_0000393.jpg',
 '9999975_00000_d_0000015.jpg',
 '9999977_00000_d_0000101.jpg',
 '9999985_00000_d_0000020.jpg',
 '9999985_00000_d_0000020.jpg',
 '9999994_00000_d_0000049.jpg',
 '9999994_00000_d_0000072.jpg',
 '9999999_00590_d_0000267.jpg']`

My question is, should we not make the pixel zero-based? This introduces a few more bounding boxes which are just lines.

Thanks and Regards,
Karthik

**Comment by zhpmatrix** [comment]
I agree with you. Actually, **boundary ground truth** will also make mistakes besides the problem you find. Good luck!

**Comment by Karthik-Suresh93** [comment]
Did you get a reply from the authors of the dataset? Also, what map are you
getting on the model you're training?

**Comment by zhpmatrix** [comment]
Yes. They invited me to join the YoMail, but I can't join successfully. That's to say, I don't get the formal reply until now. But it doesn't matter.

I'am training model named Faster R-CNN, one of my friends are trying to use Mask R-CNN.

**Comment by Karthik-Suresh93** [comment]
What is the mAP you are getting with Faster-RCNN?

**Comment by zhpmatrix** [comment]
Now, I have got the model with **Average Precision  (AP) @[ IoU=0.50:0.95 | maxDets=500 ] = 12.56%.**

I got a model with **Average Precision  (AP) @[ IoU=0.50:0.95 | maxDets=500 ] = 15.05%.** before, but I can't find it. it's so sad!

**Comment by Karthik-Suresh93** [comment]
One last question, which repo are you running faster-rcnn from?

**Comment by zhpmatrix** [comment]
https://github.com/jwyang/faster-rcnn.pytorch

**Comment by Karthik-Suresh93** [comment]
Thank you very much for your help :) I will close this issue for now. Please do open it up if you find other bugs in the dataset

**Comment by zhpmatrix** [comment]
OK.

**Comment by Karthik-Suresh93** [comment]
Hi @zhpmatrix ,

I removed the three images with xmin >xmax as you suggested. The model trained but I still faced some nan issues and got a low mAP score. Then I checked if any bounding box is outside the image, i.e 

` if xmax>img_width or ymax>img_height:   #xmax= xmin+ width_bb, ymax=ymin+height_bb`  
        `count+=1`
I found 95248 annotations with this property! Did you verify this before training? Thank you in advance for your help

**Comment by VisDrone** [comment]
Dear everyone
      I am the VisDrone team. Because we resize the images and gts  (ground truth), there are some line gts. We will update the gts after the challenge. Sorry for the line gts. 
     In this case, you can first detect the line gts (ground truth)  and remove them from the training set. 
 Regards
The VisDrone team

**Comment by Karthik-Suresh93** [comment]
Thanks!

**Comment by Karthik-Suresh93** [comment]
Hi @zhpmatrix 

I have a small question in your txt2xml conversion script 

            `fout.write('\t\t\t'+'<xmin>'+line[0]+'</xmin>'+'\n')`
            `fout.write('\t\t\t'+'<ymin>'+line[1]+'</ymin>'+'\n')`
            `# pay attention to this point!(0-based)`
            `fout.write('\t\t\t'+'<xmax>'+str(int(line[0])+int(line[2])-1)+'</xmax>'+'\n')`
            `fout.write('\t\t\t'+'<ymax>'+str(int(line[1])+int(line[3])-1)+'</ymax>'+'\n')`

Why do you only make xmax and ymax 0-based? Why not xmin and ymin?? And I think the dataset is already 0-based, so you don't need to convert it to 0-based by subtracting 1 right?

**Comment by zhpmatrix** [comment]
@Karthik-Suresh93  

for example,

bbox_left = 0, bbox_top=0, bbox_width=5, bbox_height=4, that's to say 0-based as the original dataset, then what are the coordinates of the bottom-right point?

it's exactly the original dataset is 0-based that I need to minus 1. Am I right?

**Comment by Karthik-Suresh93** [comment]
Yes, I think you're right. Not completely sure though
